PrincipalController.java

```diff
@@ -1,56 +1,51 @@
 package com.example.demo;
+
 import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.http.ResponseEntity;
 import org.springframework.http.HttpStatus;
 import java.util.*;
 import javax.servlet.http.*;
-import org.springframework.http.server.reactive.ServerHttpRequest;
-import javax.servlet.Filter;
-import javax.servlet.FilterChain;
-import javax.servlet.FilterConfig;
-import javax.servlet.ServletException;
-import javax.servlet.ServletRequest;
-import javax.servlet.ServletResponse;
-
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.*;
 import java.io.*;
 import org.springframework.stereotype.Component;
 @RestController
 
 public class PrincipalController {
-	ParkingLot pl1 = new ParkingLot();
+	@Autowired
+	ParkingLot pl1;
 	@RequestMapping()
 	String sayHello() {
 		return "Hello World!";
 	}
-	@RequestMapping(value="/park/id")
-	String park(@RequestParam("id") int carID,HttpServletResponse response) throws IOException{
-		Car car1 = new Car(carID);
+	@RequestMapping(value="/parking-lot/car/car-id/{car_id}/park")
+	String park(@PathVariable int car_id,HttpServletResponse response) throws IOException{
+		Car car1 = new Car(car_id);
 		int slotNo = car1.park(pl1);
 		if(slotNo==-1) {
 			response.sendError(HttpStatus.BAD_REQUEST.value());
 		}
-		return  "Car "+carID+" parked at "+slotNo;
+		return  "Car "+car_id+" parked at "+slotNo;
 	}
-	@RequestMapping(value="/unpark/slot")
-	String unpark(@RequestParam("slot") int slotNo,HttpServletResponse response) throws IOException{
-		Car unparked = pl1.unPark(slotNo);
+	@RequestMapping(value="/parking-lot/slot/slot-no/{slot_no}/unpark")

```

Comment:
I prefer camel case for url definitions as we can avoid usage of special characters `-` -> `parking-lot` to `parkingLot`